agent/taskresource/asmsecret/asmsecret.go

```diff
@@ -315,7 +315,7 @@ func (secret *ASMSecretResource) retrieveASMSecretValue(apiSecret apicontainer.S
 
 	secretValue, err := asm.GetSecretFromASMWithInput(input, asmClient, jsonKey)
 	if err != nil {
-		errorEvents <- err
+		errorEvents <- fmt.Errorf("fetching secret data from AWS Secrets Manager in region %s: %v", apiSecret.Region, err)
```

Comment:
nit: It's better to use `%w` format specifier over `%v` when wrapping errors as with former the underlying error can still be caught using functions like [errors.Is()](https://pkg.go.dev/errors#Is).

Reply:
good to know, thanks!